worker/src/instagram.rs

```diff
@@ -0,0 +1,81 @@
+use crate::utils::extract_signature;
+use anyhow::{anyhow, Result};
+use reqwest::header::{HeaderMap, AUTHORIZATION};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct User {
+    pub username: String,
+    pub id: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Post {
+    pub sig: String,
+    pub sig_target: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CaptionWrapper {
+    pub caption: String,
+    pub permalink: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PostWrapper {
+    pub data: Vec<PostId>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PostId {
+    pub id: String,
+}
+
+pub async fn retrieve_post(handle: String, token: String) -> Result<(String, String, String)> {
+    let url = format!(
+        "https://graph.instagram.com/me?fields=id,username&access_token={}",
+        token
+    );
+
+    let client = reqwest::Client::new();
+    let res: User = client.get(Url::parse(&url)?).send().await?.json().await?;
+
+    if handle != res.username {
```

Comment:
Should probably call `.lowercase()` on them before comparison.

Reply:
👍 on the lower case.

On the weird UX:
 
If I just redirected inside the `dapp`, how do I maintain the connection to the Beacon Wallet? Everything else could be stored in a cookie / local storage.

I'm also curious if Instagram gets angry if Domain 1 (`tzprofiles.com`) requests an authorization token and Domain 2 (`tzprofiles_witness.rebase-verifiers.workers.dev`) uses it. The worker has to be used though, cause it's got to sign the VC.

I could just stick with the worker, and divvy it up where the worker grabs the post on getting the Auth token, then, put in the KV store with the handle as the key. Then the `dapp` could send the handle+PK to the worker, which pulls the post out with the handle, and checks the sig with the PK. 

I like that last option the best because it keeps the domain getting auth'd consistent and the user just has to push a button once they've posted the caption / comment. Does that make sense?

Reply:
> If I just redirected inside the dapp, how do I maintain the connection to the Beacon Wallet? Everything else could be stored in a cookie / local storage.

Didn't think about the wallet :confused: Could it be a popup?

> I'm also curious if Instagram gets angry if Domain 1 (tzprofiles.com) requests an authorization token and Domain 2 (tzprofiles_witness.rebase-verifiers.workers.dev) uses it. The worker has to be used though, cause it's got to sign the VC.

I don't see why it would complain about the use of different IPs, but who knows. Also, the witness is behind tzprofiles.com as well.

> I could just stick with the worker, and divvy it up where the worker grabs the post on getting the Auth token, then, put in the KV store with the handle as the key. Then the dapp could send the handle+PK to the worker, which pulls the post out with the handle, and checks the sig with the PK.
> I like that last option the best because it keeps the domain getting auth'd consistent and the user just has to push a button once they've posted the caption / comment. Does that make sense?

Yeah, or issue directly and have the dapp retrieve it. That would probably be better for the user, but still doesn't feel great technically.